transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/view/AddTransferMethodFragment.java

```diff
@@ -411,18 +412,34 @@ public void showTransactionInformation(@NonNull final List<Fee> fees,
         if (isFeeAvailable(fees) && isProcessingTimeAvailable(processingTime)) {
             String formattedFee = FeeFormatter.getFormattedFee(header.getContext(), fees);
             feeAndProcessingTime.setVisibility(View.VISIBLE);
-            feeAndProcessingTime.setText(
-                    feeAndProcessingTime.getContext().getString(R.string.feeAndProcessingTimeInformation, formattedFee,
-                            processingTime.getValue()));
+            if (isZeroFeeAvailable(feeAndProcessingTime.getContext(), formattedFee)) {
```

Comment:
This method has now become too complicated with so many if/else conditions, I recommend refactoring it.
Hint: Perhaps you can handle fee and processing time in separate conditions rather than mixing them up and then call `feeAndProcessingTime.setText` at the end when you have final value to be displayed

Reply:
move to common method in FeeFormatter as getFormattedFeeAndProcessingTime()